src/mbgl/style/layers/custom_drawable_layer.cpp

```diff
@@ -181,6 +183,61 @@ class FillDrawableTweaker : public gfx::DrawableTweaker {
     float opacity;
 };
 
+class SymbolDrawableTweaker : public gfx::DrawableTweaker {
+public:
+    SymbolDrawableTweaker(const CustomDrawableLayerHost::Interface::SymbolOptions& options_)
+        : options(options_) {}
+    ~SymbolDrawableTweaker() override = default;
+
+    void init(gfx::Drawable&) override{};
+
+    void execute(gfx::Drawable& drawable, const PaintParameters& parameters) override {
+        if (!drawable.getTileID().has_value()) {
+            return;
+        }
+
+        const UnwrappedTileID tileID = drawable.getTileID()->toUnwrapped();
+        mat4 tileMatrix;
+        parameters.state.matrixFor(/*out*/ tileMatrix, tileID);
+
+        const auto matrix = LayerTweaker::getTileMatrix(
+            tileID, parameters, {{0, 0}}, style::TranslateAnchorType::Viewport, false, false, drawable, false);
+
+        static const StringIdentity idDrawableUBOName = stringIndexer().get("CustomSymbolIconDrawableUBO");
```

Comment:
You will have to update the UBO definition, after https://github.com/maplibre/maplibre-native/pull/1980 will be merged.